debezium-core/src/main/java/io/debezium/pipeline/signal/channels/DatabaseSignalChannel.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.signal.channels;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.kafka.connect.data.Struct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.annotation.NotThreadSafe;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.pipeline.signal.SignalRecord;
+
+/**
+ * The class responsible for processing of signals delivered to Debezium via a dedicated signaling table.
+ * The processor supports a common set of signals that it can process and every connector can register its own
+ * additional signals.
+ * The signalling table must conform to the structure
+ * <ul>
+ * <li>{@code id STRING} - the unique identifier of the signal sent, usually UUID, can be used for deduplication</li>
+ * <li>{@code type STRING} - the unique logical name of the code executing the signal</li>
+ * <li>{@code data STRING} - the data in JSON format that are passed to the signal code
+ * </ul>
+ *
+ * @author Jiri Pechanec
+ *
+ */
+@NotThreadSafe
+public class DatabaseSignalChannel implements SignalChannelReader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseSignalChannel.class);
+    public static final String CHANNEL_NAME = "database";
```

Comment:
Could you please use channel name `source` so it is symmetrical with `sink` notification?